### PR TITLE
GAWB-3219 Adding GUID resolution to UI

### PR DESCRIPTION
**Pull request by UrsaStutsman**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by aweng98** [comment]
Time to run WorkspaceSpec is very long. New tests adds more time for firecloud-ui-fiab-test-runner to finish because tests in same test file are run sequentially. Would you consider a new Spec test file instead adding them in WorkspaceSpec?

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Modal.scala [hunk 1]
```diff
@@ -77,5 +77,15 @@ class SynchronizeMethodAccessModal(id: String)(implicit webDriver: WebDriver) ex
 
 }
 
+class PreviewModal(id: String)(implicit webDriver: WebDriver) extends OKCancelModal(id) {
+  def getBucket = testId("Google Bucket").webElement.findElement(By.xpath("..")).getText()
+  def getObject = testId("Object").webElement.findElement(By.xpath("..")).getText()
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Modal.scala]: alternative to `..` is `"//*[@data-test-id='Google Bucket']/parent::node()"`. both works, one is shorter.

File: src/cljs/main/broadfcui/common.cljs [hunk 2]
```diff
@@ -117,6 +119,15 @@
         {:bucket-name (matcher 1)
          :object (matcher 2)}))))
 
+(defn dos-or-gcs-uri? [raw-uri]
+  (when (string? raw-uri)
+    (cond
+      (string/starts-with? raw-uri "dos://")
```
Comment (MatthewBemis) [inline src/cljs/main/broadfcui/common.cljs:125]: style nit: condition and expression can go on same line

File: src/cljs/main/broadfcui/common/gcs_file_preview.cljs [hunk 3]
```diff
@@ -128,7 +129,36 @@
...
+(react/defc DOSPreviewDialog
+  {:component-will-mount
+   (fn [{:keys [state props]}]
+     (let [{:keys [dos-uri]} props]
+       (swap! state assoc :translating-dos? true)
+       (ajax/call-martha (utils/->json-string {:url dos-uri, :pattern "gs://"})
+         {:on-done (fn [{:keys [success? raw-response xhr status-code get-parsed-response]}]
+                     (swap! state assoc
+                       :translating-dos? false
+                       :loading? true
```
Comment (MatthewBemis) [inline src/cljs/main/broadfcui/common/gcs_file_preview.cljs:142]: is `loading?` used anywhere?

File: src/cljs/main/broadfcui/common/gcs_file_preview.cljs [hunk 4]
```diff
@@ -128,7 +129,36 @@
                                  (when-not (string/blank? (@refs "preview"))
                                    (aset (@refs "preview") "scrollTop" (aget (@refs "preview") "scrollHeight"))))))})))})
 
-
+(react/defc DOSPreviewDialog
+  {:component-will-mount
+   (fn [{:keys [state props]}]
+     (let [{:keys [dos-uri]} props]
+       (swap! state assoc :translating-dos? true)
```
Comment (MatthewBemis) [inline src/cljs/main/broadfcui/common/gcs_file_preview.cljs:137]: i believe this can be set in the initial state of the component

**Review by MatthewBemis — COMMENTED** [review]
a few minor comments. code looks good otherwise.

File: src/cljs/main/broadfcui/common.cljs (hunk 2 above)
Comment (UrsaStutsman) [reply src/cljs/main/broadfcui/common.cljs:125]: I thought this made it a bit more readable. But I can change it if you've got strong opinions

File: src/cljs/main/broadfcui/common/gcs_file_preview.cljs (hunk 3 above)
Comment (UrsaStutsman) [reply src/cljs/main/broadfcui/common/gcs_file_preview.cljs:142]: Yes in the preview modal above, "loading" and "translating" are two different things, both need to happen. After translating is done via Martha, loading starts from Google

File: src/cljs/main/broadfcui/common.cljs (hunk 2 above)
Comment (MatthewBemis) [reply src/cljs/main/broadfcui/common.cljs:125]: i would have had a stronger opinion if we had done it in a consistent manner already, but it looks like we mix the two styles so ¯\\\_(ツ)_/¯

**Comment by UrsaStutsman** [comment]
jenkins retest

**Comment by UrsaStutsman** [comment]
jenkins retest

**Pull request merged**